app/api/search/entitiesIndex.js

```diff
@@ -184,10 +184,15 @@ const equalPropMapping = (mapA, mapB) => {
 
   const sameAmountOfProps =
     Object.keys(mapA.properties).length === Object.keys(mapB.properties).length;
-  const sameProps = Object.keys(mapA.properties).reduce(
-    (result, propKey) => result && mapA.properties[propKey].type === mapB.properties[propKey].type,
-    true
-  );
+  const sameProps =
+    Object.keys(mapA.properties).length === Object.keys(mapB.properties).length &&
+    Object.keys(mapA.properties).reduce(
+      (result, propKey) =>
+        result &&
+        mapB.properties[propKey] &&
+        mapA.properties[propKey].type === mapB.properties[propKey].type,
+      true
+    );
   return sameAmountOfProps && sameProps;
```

Comment:
@konzz maybe you forgot to remove the constant sameAmountOfProps in the two places or use this value in the condition to avoid executing the condition and the methods inside twice.

Reply:
Yeah, good one